src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -88,7 +89,11 @@
 public class BourneShellScriptTest {
     @Parameters(name = "{index}: {0}")
     public static Object[] data() {
-        return TestPlatform.values();
+        if (Objects.equals(System.getenv().get("SKIP_BINARY_GENERATION"), "true")) {
```

Comment:
Should we use system properties instead of environment variables? Or is there no easy way for the PCT to specify system properties?

I would use `BourneShellScript.class.GetName() + ". SKIP_BINARY_GENERATION"` as the property/env var name for better traceability wherever it needs to be used.

Reply:
Would the system property be able to work in the pom  as well? The pom uses that  variable to skip the  binary generation.

Reply:
I added a `TEST_BINARY` system property that gets added in the `generate-binaries-profile` profile of the pom. I was able to confirm that the system property gets applied properly [in a test PR here](https://ci.jenkins.io/job/Plugins/job/durable-task-plugin/job/PR-122/1/testReport/org.jenkinsci.plugins.durabletask/BourneShellScriptTest/). Normally there are 7 platforms to test, but when `TEST_BINARY` is false, the `UBUNTU_NO_BINARY` platform gets removed, and the `binaryCaching` unit tests get skipped.